src/machines/java/com/enderio/machines/data/recipes/SagMillRecipeProvider.java

```diff
@@ -14,33 +15,303 @@
+            	output(EIOTags.Items.DUSTS_SULFUR, 1, 0.1f, true)),
+        		BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("coal_ore", Ingredient.of(COAL_ORE), List.of(
+        		output(COAL, 3),
+            	output(COAL_POWDER.get(), 0.6f),
+            	output(DIAMOND, 0.005f),
+            	output(COBBLESTONE, 0.15f)),
+        		BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("redstone_ore", Ingredient.of(REDSTONE_ORE), List.of(
+        		output(REDSTONE, 8),
+            	output(REDSTONE, 0.2f),
+            	output(SILICON.get(), 0.8f),
+            	output(COBBLESTONE, 0.15f)
+            	), BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("diamond_ore", Ingredient.of(DIAMOND_ORE), List.of(
+        		output(DIAMOND, 2),
+            	output(DIAMOND, 0.25f),
+            	output(COAL_POWDER.get(), 0.1f),
+        		output(COBBLESTONE, 0.15f)
+        		),BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("emerald_ore", Ingredient.of(EMERALD_ORE), List.of(
+        		output(EMERALD, 2),
+            	output(EMERALD, 0.25f),
+        		output(COBBLESTONE, 0.15f)
+        		),BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("lapis_ore", Ingredient.of(LAPIS_ORE), List.of(
+        		output(LAPIS_LAZULI, 8),
+            	output(LAPIS_LAZULI, 0.2f),
+        		output(COBBLESTONE, 0.15f)
+        		),BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("quartz_ore", Ingredient.of(NETHER_QUARTZ_ORE), List.of(
+        		output(QUARTZ, 2),
+            	output(QUARTZ_POWDER.get(), 0.1f),
+        		output(NETHERRACK, 0.15f)
+        		),BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("quartz", Ingredient.of(QUARTZ), List.of(
+        		output(QUARTZ_POWDER.get(), 1),
+            	output(QUARTZ_POWDER.get(), 0.1f)
+        		),BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("wheat", Ingredient.of(WHEAT), List.of(
+        		output(FLOUR.get()),
+        		output(Tags.Items.SEEDS_WHEAT, 1, 0.2f, true)),
+        		BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("cobblestone", Ingredient.of(Tags.Items.COBBLESTONE), List.of(
+        		output(GRAVEL, 0.7f), 
+        		output(GRAVEL, 0.3f), 
+        		output(SAND, 0.1f), 
+        		output(FLINT, 0.05f)), 
+        		BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("sandstone", Ingredient.of(Tags.Items.SANDSTONE), List.of(
+        		output(SAND, 2), 
+        		output(SAND, 2, 0.4f)), 
+        		BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("sand", Ingredient.of(SAND), List.of(
+        		output(SILICON.get(), 0.5f)
+        		),BASE_ENERGY_PER_OPERATION, finishedRecipeConsumer);
+        
+        build("glowstone", Ingredient.of(GLOWSTONE), List.of(//swap with any:glowstone
```

Comment:
What do you mean any:glowstone, you mean a tag for any glowstone block?

Reply:
Yes, It'd be for like the chisel variants of glowstone blocks

Reply:
that is chisel:glowstone, but I'm not sure if we should have such a tag. Maybe we should create a tag like forge:glowstone and put chisel:glowstone into it

Reply:
But I didn't find any tag, so leaving it as it is for now...

Reply:
Creating tags are far above my knowledge, so I'm afraid that has to be left out of this PR